tmk_core/protocol/usb_descriptor.c

```diff
@@ -319,6 +319,20 @@ const USB_Descriptor_HIDReport_Datatype_t PROGMEM SharedReport[] = {
         HID_RI_REPORT_SIZE(8, 16),
         HID_RI_INPUT(8, HID_IOF_DATA | HID_IOF_ARRAY | HID_IOF_ABSOLUTE),
     HID_RI_END_COLLECTION(0),
+
+    HID_RI_USAGE_PAGE(8, 0x01),           // Generic Desktop
+    HID_RI_USAGE(8, 0x0C),                // Wireless Radio Controls
+    HID_RI_COLLECTION(8, 0x01),           // Application
+        HID_RI_REPORT_ID(8, REPORT_ID_RADIO),
+        HID_RI_LOGICAL_MINIMUM(8, 0x00),
+        HID_RI_LOGICAL_MAXIMUM(8, 0x01),
+        HID_RI_USAGE(8, 0xC6),            // Wireless Radio Button
+        HID_RI_REPORT_COUNT(8, 1),
+        HID_RI_REPORT_SIZE(8, 1),
+        HID_RI_INPUT(8, HID_IOF_DATA | HID_IOF_VARIABLE | HID_IOF_RELATIVE),
+        HID_RI_REPORT_SIZE(8, 7),
+        HID_RI_INPUT(8, HID_IOF_CONSTANT | HID_IOF_VARIABLE | HID_IOF_ABSOLUTE),
+    HID_RI_END_COLLECTION(0),
```

Comment:
This is almost identical to the example in https://www.usb.org/sites/default/files/hutrr40radiohidusagesfinal_0.pdf, so should be good.

However, do we want to have an option to disable this feature while keeping the existing extrakey support?  I wonder whether unconditional addition of this new HID application collection could break things like #19472 for some users.

Getting the Wireless Radio LED status could be interesting too, if some OSes actually send that data, but handling the output reports for that would need some work.

Reply:
> However, do we want to have an option to disable this feature while keeping the existing extrakey support?

Yes, might be a good idea, also because most people probably don't need it.